OpenDreamServer/Net/DreamConnection.cs

```diff
@@ -209,7 +209,7 @@ class DreamConnection {
                     promptWaitHandle.Set();
                 };
 
-                SendPacket(new PacketPrompt(promptId, types, message));
+                SendPacket(new PacketPrompt(promptId, types, "NonNull", message));
```

Comment:
This is specifically where I'm having trouble - how to get the title set to this point?

Reply:
`input()` works using an opcode called `Prompt`. If you want to change its behavior, you do that [here](https://github.com/wixoaGit/OpenDream/blob/55efe282984927cc27981525a715cbbf2eeb2722/OpenDreamServer/Dream/Procs/DreamProcInterpreterOpcodes.cs#L1087).

Line 1094 is where it tries to understand arguments `input()` was given. Since title is the third argument if you give the recipient, and the second if not, you'll want to do something like
`title = arguments.OrderedArguments[2].GetValueAsString();` in the if, and
`title = arguments.OrderedArguments[1].GetValueAsString();` in the else.

Once you get the title from the arguments you can pass this to the Prompt method on line 1104.